### PR TITLE
COMP: reuse MembersGenerator in RsImplTraitMemberCompletionProvider

### PR DESCRIPTION
**Pull request by Kobzol**
Improves https://github.com/intellij-rust/intellij-rust/pull/6342 by reusing the `MembersGenerator` class.

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/refactoring/implementMembers/impl.kt [hunk 1]
```diff
@@ -190,26 +189,38 @@ private class MembersGenerator(
...
+            else ->
+                error("Unknown trait member")
+        }
+    }
+
     private fun RsFunction.renderFunctionSignature(): String = renderer.renderFunctionSignature(this)
     private fun RsTypeReference.renderTypeReference(): String = renderer.renderTypeReference(this)
 }
+
+private fun String.withSpace(): String {
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/refactoring/implementMembers/impl.kt:220]: Hmm, when is it possible that `renderFunctionSignature` ends with space?

File: src/main/kotlin/org/rust/lang/core/completion/RsImplTraitMemberCompletionProvider.kt [hunk 2]
```diff
@@ -128,22 +108,18 @@ private fun completeType(target: RsTypeAlias): LookupElementBuilder {
 private fun completeFunction(
     target: RsFunction,
     impl: RsImplItem,
-    substitution: RsPsiSubstitution
+    substitution: RsPsiSubstitution,
+    memberGenerator: MembersGenerator
 ): LookupElementBuilder {
-    val renderer = ImportingPsiRenderer(PsiRenderingOptions(renderGenericsAndWhere = true), substitution, impl)
     val shortRenderer = ImportingPsiRenderer(PsiRenderingOptions(renderGenericsAndWhere = false), substitution, impl)
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/completion/RsImplTraitMemberCompletionProvider.kt:114]: By the way, `ImportingPsiRenderer` looks too heavy for the completion presentation. Maybe it's better to use simple `PsiSubstitutingPsiRenderer` (with default `PsiRenderingOptions.shortPaths = true` option)?

**Review by vlad20012 — APPROVED** [review]

File: src/main/kotlin/org/rust/ide/refactoring/implementMembers/impl.kt (hunk 1 above)
Comment (Kobzol) [reply src/main/kotlin/org/rust/ide/refactoring/implementMembers/impl.kt:220]: Oh, it's not. I was just surprised that the space was added automatically in implement members fix, but not here in completion. Probably in implement members the auto formatter is run automatically.

File: src/main/kotlin/org/rust/lang/core/completion/RsImplTraitMemberCompletionProvider.kt (hunk 2 above)
Comment (Kobzol) [reply src/main/kotlin/org/rust/lang/core/completion/RsImplTraitMemberCompletionProvider.kt]: Good point, using the importing renderer in presentation is probably not a good idea. I changed it per your suggestion.

**Comment by Kobzol** [comment]
bors r=vlad20012

**Pull request merged**